### PR TITLE
ROC-4698/ff4j admissions failure alerts

### PR DESCRIPTION
**Pull request by tharacka**
set up alerts for ff4j cmc_admissions failures from CMC

### CODE REVIEW

File: alerts.tf [hunk 1]
```diff
@@ -41,3 +41,21 @@ AIQ
...
+  app_insights_name = "${azurerm_application_insights.appinsights.name}"
+
+  alert_name = "ff4j response failure - CMC"
+  alert_desc = "Triggers when a ff4J cmc_admissions failure event is received from CMC in a 5 minute poll."
+  app_insights_query = "customEvents | where name == \"ff4J cmc_admissions failure\""
+  frequency_in_minutes = 5
+  time_window_in_minutes = 5
+  severity_level = "3"
+  action_group_name = "${module.cmc-ff4j-admissions-fail-action-group.action_group_name}"
+  custom_email_subject = "CMC Bulk Print Failure"
```
Comment (andrewwa-kainos) [inline alerts.tf]: Update the email subject

**Review by andrewwa-kainos — CHANGES_REQUESTED** [review]

File: action-groups.tf [hunk 2]
```diff
@@ -43,3 +43,26 @@ module "cmc-pdf-fail-action-group" {
...
+  name = "ff4j-admissions-failure-email"
+  vault_uri = "${data.azurerm_key_vault.cmc_key_vault.vault_uri}"
+}
+
+module "cmc-ff4j-admissions-fail-action-group" {
+  source = "git@github.com:hmcts/cnp-module-action-group"
+  location = "global"
+  env = "${var.env}"
+
+  resourcegroup_name = "${azurerm_resource_group.rg.name}"
```
Comment (markry-kainos) [inline action-groups.tf:59]: minor: `resource_group_name`

**Review by markry-kainos — APPROVED** [review]

File: alerts.tf [hunk 3]
```diff
@@ -41,3 +41,21 @@ AIQ
...
+  alert_desc = "Triggers when a ff4J cmc_admissions failure event is received from CMC in a 5 minute poll."
+  app_insights_query = "customEvents | where name == \"ff4J cmc_admissions failure\""
+  frequency_in_minutes = 5
+  time_window_in_minutes = 5
+  severity_level = "3"
+  action_group_name = "${module.cmc-ff4j-admissions-fail-action-group.action_group_name}"
+  custom_email_subject = "CMC FF4J CMC_Admissions Failure"
+  trigger_threshold_operator = "GreaterThan"
+  trigger_threshold = 0
+  resourcegroup_name = "${azurerm_resource_group.rg.name}"
```
Comment (andrewwa-kainos) [inline alerts.tf:60]: please use snake case

**Review by andrewwa-kainos — CHANGES_REQUESTED** [review]

File: action-groups.tf (hunk 2 above)
Comment (tharacka) [reply action-groups.tf:59]: this is used on all the definitions and these definitions are used in the action-groups.tf. Do you want me to change all of these instances? If I do, will it cause any issues with existing alerts?

Comment (markry-kainos) [reply action-groups.tf:59]: sorry Tharack - this is a module variable anyway so we can't change !!! https://github.com/hmcts/cnp-module-action-group

**Review by markry-kainos — APPROVED** [review]

**Review by andrewwa-kainos — APPROVED** [review]

**Comment by markry-kainos** [comment]
@tharacka added vault keys to `demo, aat, prod, saat, sprod`

non prod: `civilmoneyclaims+ff4j-admissions-failure@gmail.com`
prod: `cmc-live-support@hmcts.net`

**Pull request merged**